src/features/bivariate_color_manager/components/ColorLegendFilters/ColorLegendFilters.tsx

```diff
@@ -0,0 +1,231 @@
+          sentimentFilterValues.current.splice(index, 1);
+          setSentimentFiltersCount(sentimentFiltersCount - 1);
+        } else {
+          // when resetting first filter.
+          // if sentiment filters count is 2 (actual filter and plus button (which is also a filter but without a
+          // selected value)). Then set filters count to 1 and reset it's value, so plus button/filter will be removed,
+          // only first actual filter without value will be shown
+          if (sentimentFiltersCount === 2) {
+            setSentimentFiltersCount(1);
+            sentimentFilterValues.current[0].value = [];
+          } else {
+            // if amount of filters is more than 2. remove first filter and shift values
+            sentimentFilterValues.current.shift();
+            setSentimentFiltersCount(sentimentFiltersCount - 1);
+          }
+        }
+        setSentimentsFilter(
+          sentimentFilterValues.current.reduce((acc, sf) => {
+            if (sf.value.length) {
+              acc.push(sf.value);
+            }
+            return acc;
+          }, [] as string[][]),
+        );
+      }
+    },
+    [sentimentFiltersCount, setSentimentFiltersCount, setSentimentsFilter],
+  );
+
+  const onSelectSentiment = useCallback(
+    (
+      selection: SelectItemType | SelectItemType[] | null | undefined,
+      index: number,
+    ) => {
+      if (selection && Array.isArray(selection) && selection?.length) {
+        if (
+          !sentimentFilterValues.current[index].value ||
+          !sentimentFilterValues.current[index].value.length
+        ) {
+          sentimentFilterValues.current.push({ key: nanoid(4), value: [] });
+          setSentimentFiltersCount(sentimentFiltersCount + 1);
+        }
+        sentimentFilterValues.current[index].value = selection?.map((itm) =>
+          itm.value.toString(),
+        );
+        setSentimentsFilter(
+          sentimentFilterValues.current.reduce((acc, sf) => {
+            if (sf.value.length) {
+              acc.push(sf.value);
+            }
+            return acc;
+          }, [] as string[][]),
+        );
+
+        if (selectedLayer) {
+          setSelectedLayer(undefined);
+          setLayersFilter();
+        }
+      } else {
+        onResetSentiments(index);
+      }
+    },
+    [
+      sentimentFiltersCount,
+      setSentimentFiltersCount,
+      onResetSentiments,
+      selectedLayer,
+      setSelectedLayer,
+      setLayersFilter,
+      setSentimentsFilter,
+    ],
+  );
+
+  const onChangeLayer = useCallback(
+    (item) => {
+      if (item.selectedItem && sentimentFilterValues.current[0].value?.length) {
+        if (sentimentFiltersCount > 1) {
+          setSentimentFiltersCount(1);
+        }
+        sentimentFilterValues.current.length = 1;
+        sentimentFilterValues.current[0].value = [];
+        setSentimentsFilter();
+      }
+      setSelectedLayer(item.selectedItem?.value || undefined);
```

Comment:
I see that we have duplicated state in React state and atom state
Do we really need local state? 

If no, we could remove 
```
  const [selectedLayer, setSelectedLayer] = useState<
    AutocompleteItemType['value'] | undefined
  >(undefined);
```
and instead of selectedLayer use data from atom